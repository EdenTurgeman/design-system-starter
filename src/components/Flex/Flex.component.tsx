import React, { FunctionComponent } from 'react';
import { flexbox, FlexboxProps } from 'styled-system';
import styled from 'styled-components';

export interface FlexProps extends FlexboxProps {
  gap?: number;
}

const StyledFlex = styled.div<FlexProps>`
  display: flex;
  ${flexbox};
  gap: ${({ gap }) => gap}px;
`;

export const Flex: FunctionComponent<FlexProps> = (props: any) => <StyledFlex {...props} />;
