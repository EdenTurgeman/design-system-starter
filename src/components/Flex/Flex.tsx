import React, { FunctionComponent } from 'react';
import { flex, FlexProps as StyledSystemFlex } from 'styled-system';
import styled from 'styled-components';

interface FlexProps extends StyledSystemFlex {
}

const StyledFlex = styled.div<FlexProps>`
    display: flex;
    width: 200px;
    background-color: red;
    ${flex};
`;

export const Flex: FunctionComponent<FlexProps> = props  => (
  <StyledFlex {...props}/>
);
