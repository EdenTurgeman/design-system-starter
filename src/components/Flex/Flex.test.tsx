import React from 'react';
import renderer from 'react-test-renderer';
import { Flex } from './Flex.component';

test('Basic', () => {
  const component = renderer.create(<Flex>Facebook</Flex>);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Flex with gaps', () => {
  const component = renderer.create(<Flex gap={20}>Facebook</Flex>);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
