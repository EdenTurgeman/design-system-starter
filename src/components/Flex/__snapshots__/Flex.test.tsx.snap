// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Basic 1`] = `
<div
  className="sc-bdnxRM feraXx"
>
  Facebook
</div>
`;

exports[`Flex with gaps 1`] = `
<div
  className="sc-bdnxRM cOKwbR"
>
  Facebook
</div>
`;
